torch_xla/csrc/xla_sharding_util.cpp

```diff
@@ -326,13 +327,13 @@ std::vector<runtime::ComputationClient::DataPtr> ShardingUtil::OutputHandler(
     XLATensor::ShardingSpecPtr sharding = sharding_specs[i];
     if (replicated_output && sharding &&
         (sharding->sharding.type() != xla::OpSharding::REPLICATED)) {
-      XLA_CHECK(sharding->shape.has_value())
-          << "Sharding or Wrapping data shards in OutputHandler requires "
-             "unpartitioned tensor shape.";
+      // XLA_CHECK(sharding->shape.has_value())
+      //     << "Sharding or Wrapping data shards in OutputHandler requires "
+      //        "unpartitioned tensor shape.";
```

Comment:
Should this be uncommented?

Reply:
Since we have made sharding->shape no Optional now. I'm not sure what are we checking here ?

Reply:
Oh gotcha, I didn't read the whole `XLA_CHECK`, you can just delete this